src/vivarium/framework/values/manager.py

```diff
@@ -66,19 +65,21 @@ def setup(self, builder: Builder) -> None:
 
     def on_post_setup(self, _event: Event) -> None:
         """Finalizes dependency structure for the pipelines."""
-        # Unsourced pipelines might occur when generic components register
-        # modifiers to values that aren't required in a simulation.
-        unsourced_pipelines = [p for p, v in self.items() if not v.source]
-        if unsourced_pipelines:
-            self.logger.warning(f"Unsourced pipelines: {unsourced_pipelines}")
-
-        # register_value_producer and register_value_modifier record the
-        # dependency structure for the pipeline source and pipeline modifiers,
-        # respectively. We don't have enough information to record the
-        # dependency structure for the pipeline itself until now, where
-        # we say the pipeline value depends on its source and all its
-        # modifiers.
-        for pipeline in self.values():
+        for pipeline in self._all_pipelines.values():
+            # Unsourced pipelines might occur when generic components register
+            # modifiers to values that aren't required in a simulation.
+            if not pipeline.source:
+                self.logger.warning(
```

Comment:
Should we remove it from the dict completely?

Reply:
I think it's fine to keep it in there. It's also dangerous to modify the object that you're currently iterating over.